project-base/storefront/components/Layout/Header/HamburgerMenu/HamburgerMenu.tsx

```diff
@@ -4,7 +4,7 @@ import { MouseEventHandler } from 'react';
 import { twJoin } from 'tailwind-merge';
 
 type HamburgerMenuProps = {
-    onClick?: MouseEventHandler<HTMLDivElement>;
+    onClick: MouseEventHandler<HTMLDivElement>;
```

Comment:
Little improvement `onClick: MouseEventHandler<HTMLDivElement> | undefined`. This way you have to send onClick prop but you can send undefined value (which is little better than sending function which returns undefined, right?}.

Reply:
Good, refactored, thanks